src/database/DataTypes/StocktakeItem.js

```diff
@@ -315,9 +324,9 @@ export class StocktakeItem extends Realm.Object {
    * @return {bool}
    */
   get shouldHaveReason() {
-    const { snapshotTotalQuantity, countedTotalQuantity } = this;
+    const { snapshotTotalQuantity, countedTotalQuantity, hasReason } = this;
     const equalQuantities = snapshotTotalQuantity === countedTotalQuantity;
-    return !equalQuantities;
+    return !(equalQuantities || hasReason);
```

Comment:
I feel like `hasReason` isn't accurate to its name anymore and its new implementation makes `equalQuantities` redundant. So I'd just nix `hasReason` and change `shouldHaveReason` do what `hasReason` does in this PR.
```javascript
  /**
   * Returns true if the snapshot and counted quantities on any batch differ.
   * Does not account for if reasons are used by the user, caller
   * needs to account for this.
   * @return {bool}
   */
  get shouldHaveReason() {
    if (!this.batches) return false;
    return this.batches.every(({ snapshotTotalQuantity, countedTotalQuantity, option }) =>
      snapshotTotalQuantity !== countedTotalQuantity ? !!option : true
    );
  }
```

Reply:
I got wondering why you changed from `some` to `every`. Interestingly a key difference between `some` and `every` is that on an empty array the former will always return `falsy` and the latter `truthy`. So I think you could kill the early return 

```javascript
get shouldHaveReason() {
  return this.batches.some(({ snapshotTotalQuantity, countedTotalQuantity, option }) =>
    snapshotTotalQuantity !== countedTotalQuantity ? !option : false
  );
}
```

Reply:
930e6108c27d8e4bf53bbd50af875b09b4c05758 
8caa0a429e1e2844ada6d04da1797770bac010b5